library/Backends/llvm/src/Juvix/Backends/LLVM/Codegen/Record.hs

```diff
@@ -1 +1,132 @@
+        -- having to perform any recursive size calculations for now.
+        map Types.pointerOf llvmFieldTypes
+    }
+
+-- | Get the names and types of the fields of the record type with the given name.
+recordFields :: Types.Define m => PassTypes.RecordName -> m [(Symbol, Type.Type)]
+recordFields recordName = do
+  recordTable <- get @"recordTab"
+  pure $
+    snd $
+      fromMaybe (P.error "typechecker allowed record of non-existent type") $
+        Map.lookup (toSymbol recordName) recordTable
+
+-- | Get the index of the field with the given name within the list of
+-- | fields of the record with the given name.
+fieldIndex ::
+  Types.Define m =>
+  PassTypes.RecordName ->
+  PassTypes.FieldName ->
+  m Int
+fieldIndex recordName fieldName = do
+  fieldDescs <- recordFields recordName
+  pure $
+    fromMaybe (P.error "typechecker allowed selection of non-existent field") $
+      List.elemIndex (toSymbol fieldName) $ map fst fieldDescs
+
+-- | Given the name of a record type and a pointer to the location of
+-- | a record term with that type, get a pointer to the location of
+-- | the pointer to the field with the given name inside the record.
+getFieldPtr ::
+  Types.Define m =>
+  PassTypes.RecordName ->
+  PassTypes.FieldName ->
+  AST.Operand ->
+  m AST.Operand
+getFieldPtr recordName fieldName location = do
+  fieldDescs <- recordFields recordName
+  index <- fieldIndex recordName fieldName
+  Block.getElementPtr
+    Types.Minimal
+      { type' = Types.pointerOf $ Types.pointerOf $ snd $ fieldDescs P.!! index,
+        address' = location,
+        indincies' = Block.constant32List [0, toInteger index]
+      }
+
+-- | Given the name of a record type, a pointer to the location of
+-- | a record term with that type, the name of a field in the record,
+-- | and an LLVM type, load the field with the given name, treating
+-- | it as the given LLVM type.
+loadField ::
+  Types.Define m =>
+  PassTypes.RecordName ->
+  PassTypes.FieldName ->
+  AST.Operand ->
+  Type.Type ->
+  m AST.Operand
+loadField recordName fieldName location ty = do
+  fieldDescs <- recordFields recordName
+  index <- fieldIndex recordName fieldName
+  field <-
+    Block.loadElementPtr
+      Types.Minimal
+        { type' = Types.pointerOf $ snd $ fieldDescs P.!! index,
+          address' = location,
+          indincies' = Block.constant32List [0, toInteger index]
+        }
+  Block.load ty field
+
+-- | Given the name of a record type and a list of compiled field terms,
+-- | allocate a record and store the given terms in it.
+makeRecord ::
+  Types.Define m =>
+  PassTypes.RecordName ->
+  [AST.Operand] ->
+  m AST.Operand
+makeRecord recordName fieldTerms = do
+  recordTable <- get @"recordTab"
+  let (recordType, fieldDescs) = fromMaybe (P.error "typechecker allowed record of non-existent type") (Map.lookup (toSymbol recordName) recordTable)
+  recordPtr <- Block.mallocType recordType
```

Comment:
we should probably note we leak memory here for now

Reply:
Done in https://github.com/anoma/juvix/pull/1106/commits/b7e4f0593262df3ef8a2079355c5ca06bf9bd739